main.js

```diff
@@ -1,141 +1,413 @@
+                        try {
+                            setClicked(button, false);
+                            var newbutton = randomButton(button.power);
+                            newbutton.id = player.buttons.length;
+                                updateButtonStats(newbutton);
+                            player.buttons.push(newbutton);
+                            player.buttonsmade++;
+                            button.speed*=1.2;
+                            updateButtonStats(button);  
+                        } catch (error) {
+                            console.error(error);
+                        }
+                    }, button.speed / speedup);
+                    setClicked(button,true);
+                    break;
+                case "shards":
+                    setTimeout(function() {
+                        try {
+                            setClicked(button, false);
+                            //player.shards += button.power;
+                            //update("shardsbox", player.shards.toFixed(1));
+                        } catch (error) {
+                            console.error(error);
+                        }
+                    }, button.speed / speedup);
+                    button.shardPerSec=(button.power/button.speed)*1000;
+                    button.shardGain=button.power;
+                    setClicked(button,true);
+                    break;
+                case "speed":
+                    var target = player.buttons[button.target];
+                    if(button.target>-1&&target!=null) {
+                        button.shardPerSec=(target.speedCost/button.speed)*button.costmult*1000;
+                        button.shardUse+=target.speedCost*button.costmult;
+                        setClicked(button,true);
+                    }
+                    break;
+                case "power":
+                    var target = player.buttons[button.target];
+                    if(button.target>-1&&target!=null) {
+                        button.shardPerSec=(target.powerCost/button.speed)*button.costmult*1000;
+                        button.shardUse+=target.powerCost*button.costmult;
+                        setClicked(button,true);
+                    }
+                    break;
+                default:
+                    console.error("unrecognized type: " + button.type);
+                    break;
+            }
+            break;
+    }
 }
-function setClicked(button,disable){
-  if(disable){
-    button.disabled=true
-    button.element.classList.add("disabled")
-    button.time=0
-    button.timeupdate=true
-  }else{
-    button.disabled=false
-    button.element.classList.remove("disabled")
-    button.timeupdate=false
-  }
+
+function setClicked(button, disable) {
+    if (disable) {
+        button.disabled = true;
+        button.element.classList.add("disabled");
+        button.time = 0;
+        button.timeupdate = true;
+    } else {
+        switch(button.type) {
```

Comment:
This stuff probably shouldn't be in a function called `setClicked`. I'd say it should be moved into a separate function.

Reply:
which part